### PR TITLE
Feature/clean error and warning messages for arm ti compiler

### PR DESCRIPTION
**Pull request by olgierdh**
[DESCRIPTION]

Fixed all compilation warnings revealed via the TI ARM compiler.

[REVIEWERS]

@atigyi 
@DellaBitta 

[QA]

All tests passes

[RELEASE NOTES]

Fixed the code so that the CC3200 target now compiles without warnings.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,25 +9,25 @@ env:
...
       XI_MAKETARGET=""
 
-    - PRESET=POSIX_DEV
-    - PRESET=POSIX_DEV_MIN
-    - PRESET=POSIX_REL
-    - PRESET=POSIX_REL_MIN
+      #- PRESET=POSIX_DEV
+      #- PRESET=POSIX_DEV_MIN
+      #- PRESET=POSIX_REL
+      #- PRESET=POSIX_REL_MIN
```
Comment (olgierdh) [inline .travis.yml]: Yeah this branch is not yet ready to be reviewed - it is missing some executables from the CC3200 SDK it's still under the testing phase. Sorry for confusion.

File: .travis.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ before_script:
 script:
   - "set -e"
   - "cd $HOME/build/xively/xively-client-c/"
-  - "expect res/travis/travis_build.expect make PRESET=$PRESET XI_DEBUG_OUTPUT=0 $XI_MAKETARGET"
+  - "expect res/travis/travis_build.expect make MD= PRESET=$PRESET XI_DEBUG_OUTPUT=0 $XI_MAKETARGET"
```
Comment (olgierdh) [inline .travis.yml]: This is still under test, see my comment above.

File: Makefile [hunk 3]
```diff
@@ -209,15 +212,6 @@ $(LIBXIVELY)/src/%.sa:
 	$(info [clang-tidy] $(@:.sa=.c))
 	@clang-tidy --checks='clang-analyzer-*,-clang-analyzer-cplusplus*,-clang-analyzer-osx*' $(@:.sa=.c) >> static_analysis_$(NOW).log -- $(XI_CONFIG_FLAGS) $(XI_COMPILER_FLAGS) $(XI_INCLUDE_FLAGS)
 
-.PHONY: static_analysis
-static_analysis:  $(XI_SOURCES:.c=.sa)
```
Comment (atigyi) [inline Makefile]: was this removed intentionally?

File: make/mt-config/tests/mt-tests-integration [hunk 4]
```diff
@@ -6,7 +6,7 @@
 include make/mt-config/tests/mt-tests
 
 XI_ITESTS_OBJDIR := $(XI_TEST_OBJDIR)/itests
-XI_ITESTS_CFLAGS := $(XI_CONFIG_FLAGS) $(XI_COMPILER_FLAGS)
+XI_ITESTS_CFLAGS = $(XI_CONFIG_FLAGS) $(XI_COMPILER_FLAGS)
```
Comment (atigyi) [inline make/mt-config/tests/mt-tests-integration]: 👍

File: src/libxively/mqtt/logic/xi_mqtt_logic_layer_publish_q1_command.h [hunk 5]
```diff
@@ -163,8 +163,6 @@ do_mqtt_publish_q1( void* ctx /* should be the context of the logic layer */
 
     XI_CR_END();
 
-    return XI_STATE_OK;
```
Comment (atigyi) [inline src/libxively/mqtt/logic/xi_mqtt_logic_layer_publish_q1_command.h]: was this a warning by ti-compiler?

File: src/tests/utests/xi_utest_mqtt_logic_layer_subscribe.c [hunk 6]
```diff
@@ -134,7 +134,7 @@ XI_TT_TESTCASE_WITH_SETUP(
         // set the task data
         XI_ALLOC_AT( xi_mqtt_logic_task_t, task, local_state );
 
-        task->cs = 109; // this is very hakish since it depends on the code
+        task->cs = 111; // this is very hakish since it depends on the code
```
Comment (atigyi) [inline src/tests/utests/xi_utest_mqtt_logic_layer_subscribe.c]: ohmy...

File: Makefile [hunk 7]
```diff
@@ -109,28 +112,28 @@ $(XI_PROTOBUF_GENERATED)/%.pb-c.c : $(XI_PROTO_DIR)/%.proto
 $(XI_UTEST_OBJDIR)/%.o : $(XI_UTEST_SOURCE_DIR)/%.c $(XI_BUILD_PRECONDITIONS)
 	@-mkdir -p $(dir $@)
 	$(info [$(CC)] $@)
-	$(MD) $(CC) $(XI_UTEST_CONFIG_FLAGS) $(XI_UTEST_INCLUDE_FLAGS) -c $< -o $@
-	@$(CC) $(XI_UTEST_CONFIG_FLAGS) $(XI_UTEST_INCLUDE_FLAGS) -MM $< -MT $@ -MF $(@:.o=.d)
+	$(MD) $(CC) $(XI_UTEST_CONFIG_FLAGS) $(XI_UTEST_INCLUDE_FLAGS) -c $< $(XI_COMPILER_OUTPUT)
+	$(MD) $(CC) $(XI_UTEST_CONFIG_FLAGS) $(XI_UTEST_INCLUDE_FLAGS) -MM $< -MT $@ -MF $(@:.o=.d)
```
Comment (olgierdh) [inline Makefile]: Yes, that was desired since I had to debug these as well.

File: Makefile (hunk 3 above)
Comment (olgierdh) [inline Makefile]: Yes it was a duplication of the code that was already there.

File: src/libxively/mqtt/logic/xi_mqtt_logic_layer_publish_q1_command.h (hunk 5 above)
Comment (olgierdh) [inline src/libxively/mqtt/logic/xi_mqtt_logic_layer_publish_q1_command.h]: Yes, non-reachable code.

File: src/tests/utests/xi_utest_mqtt_logic_layer_subscribe.c (hunk 6 above)
Comment (olgierdh) [inline src/tests/utests/xi_utest_mqtt_logic_layer_subscribe.c]: yes....

**Pull request merged**